src/CommunityToolkit.Maui.Core/Essentials/Badge/Android/DefaultBadgeProvider.android.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Diagnostics;
+using Android.Content;
+using Android.Content.PM;
+
+namespace CommunityToolkit.Maui.ApplicationModel;
+
+class DefaultBadgeProvider : IBadgeProvider
+{
+	const string intentAction = "android.intent.action.BADGE_COUNT_UPDATE";
+
+	public void SetCount(int count)
+	{
+		if (count < 0)
+		{
+			return;
```

Comment:
Let's throw an `IndexOutOfRangeException` instead of failing silently:

```suggestion
			throw new IndexOutOfRangeException("Cannot set badge count below zero");
```

Reply:
other platforms (ios, maccatalyst) ignore negative value, so we shouldn't throw exception to be consistent between different platforms